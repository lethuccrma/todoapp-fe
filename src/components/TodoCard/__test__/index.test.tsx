import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { BrowserRouter as Router } from 'react-router-dom';

import TodoCard from '../index';
import { CardStatus } from '../../../types/ICard';

const renderer = createRenderer();

describe('<TodoCard />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(
      <Router>
        <TodoCard
          title="Todo"
          content="Content"
          categories=""
          status={CardStatus.OPEN}
          dueDate={new Date('1999-08-02T00:00:00.000Z')}
          onCompleteClick={() => {
            console.log('complete');
          }}
          onDeleteClick={() => {
            console.log('delete');
          }}
          onEditClick={() => {
            console.log('edit');
          }}
        />
      </Router>,
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
