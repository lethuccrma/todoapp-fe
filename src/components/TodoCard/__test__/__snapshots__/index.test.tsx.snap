// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoCard /> should render and match the snapshot 1`] = `
<Router
  location={
    Object {
      "hash": "",
      "key": "default",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    Object {
      "action": "POP",
      "back": [Function],
      "block": [Function],
      "createHref": [Function],
      "forward": [Function],
      "go": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <TodoCard
    categories=""
    content="Content"
    dueDate={1999-08-02T00:00:00.000Z}
    onCompleteClick={[Function]}
    onDeleteClick={[Function]}
    onEditClick={[Function]}
    status="OPEN"
    title="Todo"
  />
</Router>
`;
